var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -133,6 +127,17 @@ class IntelParallelStudio(IntelPackage):
     variant('tbb',  default=True,
             description='Install the Intel TBB libraries')
 
+    provides('daal',        when='+daal')
+    provides('ipp',         when='+ipp')
+
+    provides('mkl',         when='+mkl')
+    provides('blas',        when='+mkl')
+    provides('lapack',      when='+mkl')
+    provides('scalapack',   when='+mkl')
+
+    provides('mpi',         when='+mpi')
+    provides('tbb',         when='+tbb')
```

Comment:
I prefer setting providers after variants are defined, especially since we are conditionally setting them based on variants.